### DISCUSSION TITLE
select custom user other than root for running crons

### DISCUSSION THREAD

**Discussion by radhay-1199** [jhuckaby/Cronicle#684] [Q&A]
currently my crons are configured to be run only by a specific user group on the server, but i see cronicle always picks up root user by default to run the crons which dont have the permission. how can i define the user to be picked for running jobs?

**Comment by ian-kinner** [comment]
Create a plugin for jobs that run as a specific user:
![Screenshot 2023-11-30 at 10 58 29 AM](https://github.com/jhuckaby/Cronicle/assets/15766487/0ad4aedf-77f1-4b87-b854-809ac75867f1)

**Comment by oliviermollier** [comment]
Hello, i notice that when we use this feature we only change the userid but the script doesn't herit for GID as well. My conserne is that i have dedicated user that can start a process docker so my user is 
uid=1001(xxx) gid=1001(xxx) groups=1001(xxx),992(docker).

when i launch my script i have only uid=1001(xxx) gid=1001(xxx) groups=1001(xxx)

i think it's because in the shell_plugins we only force the uid. Anyone try to force GID as well?

**Comment by jhuckaby** [comment]
I'll add this to the TODO list for sure.

**Comment by nycalex** [comment]
Hello,
Any updates on this?  Full delegation to non-privileged user to run a script is very useful, plus important security consideration. Thanks!!!

**Reply by jhuckaby** [reply]
I just released [Cronicle v0.9.45](https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.45) which has the requested feature.

![Screenshot 2024-03-12 at 10 59 44 PM](https://github.com/jhuckaby/Cronicle/assets/700765/f1f41159-8f4a-4d62-9012-d2187f62ccdc)

Upgrading Guide: https://github.com/jhuckaby/Cronicle/blob/master/docs/CommandLine.md#upgrading-cronicle

Enjoy!
